### PR TITLE
feat(mergedmining): use longpoll on bitcoin.getblocktemplate

### PR DESCRIPTION
**Pull request by jansegre**
- [x] change base to `dev` after `feat/merged-mining-better-api` is merged

### CODE REVIEW

**Comment by jansegre** [comment]
In draft because it depends on #114

File: hathor/merged_mining/coordinator.py [hunk 1]
```diff
@@ -1177,49 +1177,56 @@ def update_jobs(self) -> None:
...
-        # self.log.debug('bitcoin.getblocktemplate response', res=data_log)
-        self.bitcoin_coord_job = BitcoinCoordJob.from_dict(data)
-        self.log.debug('new Bitcoin block template', height=self.bitcoin_coord_job.height)
-        await self.update_merged_block()
+        backoff = 1
+        longpoll_id = None
+        while True:
+            self.log.debug('get Bitcoin block template')
+            try:
+                backoff = 1
```
Comment (msbrogli) [inline hathor/merged_mining/coordinator.py:1210]: Should it be after `self.bitcoin_rpc.get_block_template(...)`? It seems `await asyncio.sleep(backoff)` will always wait one second.

Comment (jansegre) [reply hathor/merged_mining/coordinator.py:1210]: You're right, my mistake.

Comment (jansegre) [reply hathor/merged_mining/coordinator.py]: Fixed.

**Review by msbrogli — APPROVED** [review]

**Pull request merged**